arch/arm/core/aarch32/cortex_m/mpu/arm_mpu_v7_internal.h

```diff
@@ -151,8 +151,16 @@ static inline u32_t mpu_region_get_size(u32_t index)
  */
 static inline int is_enabled_region(u32_t index)
 {
+	/* Lock IRQs to ensure RNR value is correct when reading RASR. */
```

Comment:
I suggest adding comments to other functions which use RNR/RASR/etc without locking interrupts, that they should be called with irqs locked.